### DISCUSSION TITLE
Question on config merging / force override

### DISCUSSION THREAD

**Discussion by lfrancke** [renovatebot/renovate#28927] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub / 37.349 & main branch

### Please tell us more about your question or problem

I have a [config in the repository](https://github.com/stackabletech/commons-operator/blob/0e0d7cc17068750524ae26d104930352fcef0f39/renovate.json) that looks like this:

```
  "lockFileMaintenance": {
    "enabled": true,
    "schedule": [
      "after 5:00 and before 6:00 every weekday"
    ]
  },
```

When running renovate manually I applied this config snippet in my `config.js`:

```
force: {
        schedule: [],
        prCreation: "immediate",
        lockFileMaintenance: {
            enabled: true,
            schedule: null
        }
    }
```

And that caused `lockFileMaintenance` to fail because it'd generate a branch name called `renovate/.x` which is wrong.
It turns out that the `force` does override the entire `lockFileMaintenance` block including the `branchTopic` (should be `lock-file-maintenance` but then defaulted to the bad value).

Is this expected?
In my repository config I also only specify a subset of options so I kinda naively assumed that I can also just specify a subset in the `force` as well.

I do _now_ see this comment in the docs:

```
We recommend you avoid the force config option, if possible.

It can be used to "force" config over the top of other config or rules which might be merged later, so at times can cause confusion - especially if it's defined in Global config and overriding settings in Repository config.
```

What is the recommended way to override things like this schedule at runtime?
That said: It seems as if the schedule is ignored anyway but that seems like a different issue.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
We probably should do a "deep" merge of lockFileMaintenance and other similar objects.

In the meantime you might be able to succeed with a package rule like this:

```json
{
  "lockFileMaintenance": {
    "enabled": true
  },
  "packageFiles": [
	{
	  "matchUpdateTypes": ["lockFileMaintenance"],
	  "force": {
	    "schedule": "at any time"
	  }
	}
  ]
}
```

This will turn on lockFileMaintenance for any repo which hasn't explicitly turned it off themselves (which should be rare, because it's off by default). Then it should force the schedule to be at any time.

**Reply by lfrancke** [reply]
Thank you!
It turns out that the schedule is ignored anyway.......don't ask me why.
But I just removed my override and everything works as expected.

But yes: I did expect it to be a deep merge as the all the other config merging uses the same strategy. But it's good to know.

Thanks for your help
